src/aditiGJUN20/SumOfDigits.java

```diff
@@ -0,0 +1,42 @@
+// Find sum of all digits in given string. (Consider consecutive digits till next character as single number)
+
+package aditiGJUN20;
+
+public class SumOfDigits {
+	int getSumofAllDigits(String str) {
+		// Condition for empty String
+		if (str.isEmpty()) {
+			System.out.println("Null String");
+			return 0;
+		}
+		
+		int digitSum = 0;
+		String temp = "0"; // Earlier tried with "" and null but had thrown exception so settled with 0 as
```

Comment:
I can think of one approach instead of declaring String temp = "0", you can declare String temp ="" and handle the condition in else part which will avoid number format exception..
else {
				if(temp.length() !=0) {
					sum += Integer.parseInt(temp);
					temp = "";
				}

Reply:
Thank you @AmitaRJun20 for your insights. I've added extra condition and handled the code as you suggested. Now the code is giving correct output with `temp=""`